AppRentCar/models.py

```diff
@@ -0,0 +1,145 @@
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+from django.db import models
+from imagekit.models import ImageSpecField
+from pilkit.processors import ResizeToFill
+from django.contrib.auth.models import User
+
+
+class BaseModel(models.Model):
+    """Base model."""
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class Car(BaseModel):
+    type_engines = {
+        ('Benzyna', 'Benzyna'),
+        ('Diesel', 'Diesel'),
+        ('Hybryda', 'Hybryda'),
+        ('Elektryczny', 'Elektryczny')
+    }
+    type_transmission = {
+        ('Automatyczna', 'Automatyczna'),
+        ('Manuala', 'Manualna')
+    }
+    number_of_gears = {
+        ('5', '5'),
+        ('6', '6'),
+        ('7', '7'),
+        ('8', '8')
+    }
+    type_drives = {
+        ('Przedni', 'Przedni'),
+        ('Tylni', 'Tylni'),
+        ('4x4', '4x4')
+    }
+    type_car = {
+        ('Kombi', 'Kombi'),
+        ('Sedan', 'Sedan'),
+        ('Coupe', 'Coupe'),
+        ('Hatchback', 'Hatchback'),
+        ('Suv', 'Suv'),
+        ('Van', 'Van'),
+        ('Shooting brake', 'Shooting brake')
+    }
+    avatar = models.ImageField(upload_to='media/avatars/', blank=True, null=True)
+    avatar_thumbnail = ImageSpecField(
+        source='avatar',
+        processors=[ResizeToFill(100, 100)],
+        format='JPEG',
+        options={'quality': 80}
+    )
+    brand = models.CharField(max_length=32)
+    model = models.CharField(max_length=32)
+    cars_type = models.CharField(max_length=32, choices=type_car)
+    engine = models.CharField(max_length=32, choices=type_engines)
+    capacity = models.FloatField()
+    year = models.CharField(max_length=8)
+    number_of_seats = models.IntegerField()
+    consumption = models.CharField(max_length=32)
+    power = models.CharField(max_length=16)
+    car_mileage = models.CharField(max_length=16)
+    transmission = models.CharField(max_length=32, choices=type_transmission)
+    no_gears = models.CharField(max_length=8, choices=number_of_gears)
```

Comment:
skoro to liczba to zmieniłbym na integerfield które jest walidowane min i max validatorem